backend/spec/models/survey_subject_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe SurveySubject, type: :model do
+  describe 'relationships' do
+    it { is_expected.to have_many(:surveys).dependent(:destroy) }
+  end
+
+  describe 'presence' do
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:description) }
+  end
+
+  describe '#destroy' do
+    describe 'When there are surveys for survey subject' do
+      let!(:survey_subject) { create(:survey).survey_subject }
+
+      it 'does not destroy' do
+        survey_subject.destroy
+        expect(survey_subject.errors.full_messages.first).to match("Can't destroy survey subject with 1 surveys.")
+      end
+
+      it { expect(described_class.count).to eq(1) }
```

Comment:
this spec just test that the factory works, you should remove it or put `survey_subject.destroy` at the `before`

Reply:
Actually on line 19 I try to destroy that class and show that It is not destroyed on line 22